File: Favorites/Favorites/Cache/CacheBus.m
//
//  CacheBus.m
//  Favorites
//
//  Created by Hongyi Zheng on 16/5/12.
//  Copyright © 2016年 洪毅 郑. All rights reserved.
//

#import "CacheBus.h"

@implementation CacheBus

+ (instancetype) ins {
    static CacheBus *cacheBus = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        cacheBus = [[super allocWithZone:NULL] init];
    });
    return cacheBus;
}

- (instancetype) init {
    self = [super init];
    if (self) {
        _category = [CacheCategory new];
        _favorite = [CacheFavorite new];
    }
    return self;
}

@end

File: Favorites/Favorites/Cache/CacheBus.swift
//
//  CacheBus.swift
//  Favorites
//
//  Created by 洪毅 郑 on 16/3/6.
//  Copyright © 2016年 洪毅 郑. All rights reserved.
//

import UIKit
class CacheBus: NSObject {
    
    var favorite: CacheFavorite
    var favoriteCategory: CacheCategory
    
    override init() {
        favorite = CacheFavorite();
        favoriteCategory = CacheCategory();
        super.init()
    }
    
    class var sharedInstance: CacheBus {
        struct Static {
            static var onceToken: dispatch_once_t = 0
            static var instance: CacheBus? = nil
        }
        dispatch_once(&Static.onceToken) { () -> Void in
            Static.instance = CacheBus();
        }
        return Static.instance!
    }

}
